pkg/clusteragent/autoscaling/workload/model/metadata.go

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package model
+
+import "encoding/json"
+
+// exported for testing purposes
+const (
+	AnnotationsURLKey         = "autoscaling.datadoghq.com/url"
```

Comment:
As you already have a JSON for settings, it can simplified as a single JSON object:

```
autoscaling.datadoghq.com/custom-recommender: {"url": <string>, "settings": {...} }
```